File: data/python/SMALLEST_SUM_CONTIGUOUS_SUBARRAY_TRANSLATED.java
====================
static int fGold ( int [ ] arr , int n ) {
  int minEndingHere = Integer . MAX_VALUE ;
  int minSoFar = Integer . MAX_VALUE ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( ( minEndingHere > 0 ) && ( arr [ i ] > 0 ) ) {
      minEndingHere = arr [ i ] ;
    }
    else {
      minEndingHere += arr [ i ] ;
    }
    minSoFar = Math . min ( minSoFar , minEndingHere ) ;
  }
  return minSoFar ;
}

File: data/java/SMALLEST_SUM_CONTIGUOUS_SUBARRAY_TRANSLATED.py
====================
def f_gold ( arr , n ) :
    min_ending_here = 2147483647
    min_so_far = 2147483647
    for i in range ( n ) :
        if min_ending_here > 0 :
            min_ending_here = arr [ i ]
        else :
            min_ending_here += arr [ i ]
        min_so_far = min ( min_so_far , min_ending_here )
    return min_so_far

